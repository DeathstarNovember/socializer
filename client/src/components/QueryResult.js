import React from "react";
import { ErrorMessage, Loading } from "components";

const QueryResult = ({ loading, error, data, children }) => {
  if (loading) {
    return <Loading />;
  } else if (error) {
    return <ErrorMessage message={error.message} />;
  } else {
    return children({ data });
  }
};

export default QueryResult;
