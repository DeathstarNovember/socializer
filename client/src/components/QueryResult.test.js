import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import QueryResult from "./QueryResult";

describe("QueryResult", () => {
  it("renders correctly when loading", () => {
    const { container } = render(<QueryResult loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when error", () => {
    const { container } = render(
      <MemoryRouter>
        <QueryResult error={{ message: "Oops" }} />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when data is ready", () => {
    const { container } = render(
      <QueryResult data={{ name: "Sam" }}>
        {({ data }) => <div>{data.name}</div>}
      </QueryResult>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
