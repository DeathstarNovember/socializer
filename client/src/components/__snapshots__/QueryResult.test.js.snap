// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryResult renders correctly when data is ready 1`] = `
<div>
  Sam
</div>
`;

exports[`QueryResult renders correctly when error 1`] = `
<div
  class="d-flex justify-content-center"
>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Error
      </div>
      <p
        class="card-text"
      >
        Something went wrong, sorry about that.
      </p>
      <a
        href="/"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Go home
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`QueryResult renders correctly when loading 1`] = `
<div
  class="d-flex justify-content-center"
>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
